staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -325,6 +326,20 @@ func (a *Allocator) Allocate(ctx context.Context, node *v1.Node) (finalResult []
 				Device:      internal.id.Device.String(),
 				AdminAccess: internal.adminAccess,
 			}
+			for _, device := range internal.slice.Spec.Devices {
+				if device.Name == internal.id.Device {
+					resultAllocatedDeviceStatus = append(resultAllocatedDeviceStatus, resourceapi.AllocatedDeviceStatus{
+						Driver:                   internal.id.Driver.String(),
+						Pool:                     internal.id.Pool.String(),
+						Device:                   internal.id.Device.String(),
+						UsageRestrictedToNode:    device.Basic.UsageRestrictedToNode,
+						BindingConditions:        device.Basic.BindingConditions,
+						BindingFailureConditions: device.Basic.BindingFailureConditions,
+						BindingTimeout:           device.Basic.BindingTimeout,
+					})
+				}
+			}
+
```

Comment:
Let's add the device status only when needed by a device.

You also have to add feature gate checking: I don't remember whether it was spelled out explicitly in the KEP (if not, please add in a follow-up), but what would make sense to me is to ignore devices which have binding conditions when the feature is turned off. In other words, don't select them because the code which waits during binding wouldn't be active.

Reply:
I add allocatedDeviceStatus only when BindingConditions exist.
and I pass the featureGate status of BindingConditions to the allocator and check it.